llvm/include/llvm/CodeGen/ValueTypes.td

```diff
@@ -331,6 +331,10 @@ def i64x8     : ValueType<512,  231>;  // 8 Consecutive GPRs (AArch64)
 def aarch64svcount
               : ValueType<16,  232>;  // AArch64 predicate-as-counter
 def spirvbuiltin : ValueType<0, 233>; // SPIR-V's builtin type
+// AMDGPU buffer fat pointer, buffer rsrc + offset, rewritten before MIR translation
+def amdgpuBufferFatPointer : ValueType<160, 234>;
+// AMDGPU buffer strided pointer, buffer rsrc + index + offset, doesn't reach MIR
+def amdgpuBufferStridedPointer : ValueType<192, 235>;
```

Comment:
What's the reason to define these as opaque target constants? Why can't we just use i160? That would allow dropping the override of getPointerTy too

Reply:
Because you get a bunch of failures around things not being legal truncations and other weirder issues if you allow an i160

Reply:
Like, maybe there's a way to make this work but since no one wants to actually codegen an i160 I'm not sure I see a reason to

Reply:
Can you add a comment at least that these should just be i160?

Reply:
Fixing those probably shouldn't be that bad. Do you have an example?

Reply:
Got sucked into gfx12 bringup stuff, here's a branch that *just* add i160 and doesn't even try to touch fat pointers, observe 15 diverse crashes on the x86 and amdgpu test suite https://github.com/llvm/llvm-project/pull/127953 

I'm thinking that just making a custom type - whose size is implicitly 160 bits per the way tablegen works, makes sense, since the ARM forks added i64x8 (8 contiguous registers) as a custom type for similar reasons instead of going to MVT::i512

And ... this type never hits  MIR anyway so

Reply:
To put out one more point, from what I can tell, it's an invariant of SelectionDAG that all the `MVT::i*` are powers of two - see also the lack of an `MVT::i96` even though we actually have instructions that that could meaningfully apply to.

Reply:
It really shouldn't be an invariant, but the fact that most operations default to being legal for every type can cause problems when new types are added. I know we've had issues with extending loads in the past when new types were added.

Reply:
And since these are special-purpose MVT entries that shouldn't hit codegen (and that would need special handling if they did - if we were to do address space 7 at the MIR level, we'd wouldn't want the usual `sexts` that come with GEP) I figure this is something that makes more sense as a standalone MVT instead of trying to make i160 happen.